.github/scripts/init_linux_port_vm_network.sh

```diff
@@ -5,37 +5,85 @@
 #
 # This script sets up virtual interface (veth1 and veth0 ) with DHCP configure that can be used for linux port.
 
-ip link add veth0 type veth peer name veth1
-ifconfig veth0 192.168.1.1 netmask 255.255.255.0 up
-ifconfig veth1 up
-ethtool --offload veth0 tx off
-ethtool --offload veth1 tx off
-
-sh -c "cat > /etc/dhcp/dhcpd.conf" <<EOT
-subnet 192.168.1.0 netmask 255.255.255.0 {
-    range 192.168.1.100 192.168.1.200;
-    option routers 192.168.1.1;
-    option domain-name-servers 1.1.1.1;
-    option broadcast-address 192.168.1.255;
-    default-lease-time 600;
-    max-lease-time 7200;
+getNextAvailableIpRange() {
+    for (( i=1; i<=255; i++ )); do
+        match=`ifconfig -a | grep -o -E "inet $1\.$2\.$i\.[0-9]+"`
+        #echo "$match"
+
+        if [[ "$match" == "" ]]; then
+            echo $i
+            break
+        fi
+    done
 }
+
+if [[ "$1" == "--help" || "$1" == "-h" ]]; then
+    echo "Creates a virtual network interface for the freertos simulator."
+    echo "Use option --clean to undo any changes."
+elif [[ "$1" == "--clean" ]]; then
+    ifconfig veth0 down
+    ifconfig veth1 down
+    ip link delete veth0
+    rm -f /etc/dhcp/dhcpd.conf
+    rm -f /etc/default/isc-dhcp-server
+
+    if [ -f "/etc/dhcp/dhcpd.conf.bkp" ]; then
+        mv "/etc/dhcp/dhcpd.conf.bkp" "/etc/dhcp/dhcpd.conf"
+    fi
+
+    if [ -f "/etc/default/isc-dhcp-server.bkp" ]; then
+        mv "/etc/default/isc-dhcp-server.bkp" "/etc/default/isc-dhcp-server"
+    fi
+
+    service isc-dhcp-server restart
+
+    iptables-restore < /opt/iptables.backup
```

Comment:
is it ok to assume that this file exists /opt/iptables.backup?

Reply:
If the customer is running the --clean, it is assumed they have run the bare script before. If the file is not there it's safe because the command will fail. I'll double check that to be sure.

Reply:
Yeah, it's safe. The command fails (and the current iptables config does not get affected) if the file is not present.

```bash
ewertons@ewertons-lcx1g3:~/code/work/s1/iot-middleware-freertos-samples-ew$ sudo iptables-restore < ./test.bkp 
bash: ./test.bkp: No such file or directory 
```

Reply:
This will cause script to finish in unclean maner. Can we check if file exists.